crates/oxc_semantic/src/builder.rs

```diff
@@ -100,6 +100,39 @@ pub struct SemanticBuilderReturn<'a> {
     pub errors: Vec<OxcDiagnostic>,
 }
 
+#[derive(Default)]
+pub struct Collector {
```

Comment:
We should move this to another file and have some doc comments explaning the why before merge.